File: Backend/app/src/main/java/com/example/demo/model/Image.java
package com.example.demo.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;
import java.time.LocalDateTime;

@Entity
@Table(name = "images")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Image {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Long userId;               // Ai upload ảnh

//    private String title;

//    @Column(columnDefinition = "TEXT")
//    private String description;

    private String url;

    private String status;        // draft / processing / published / private / deleted / blocked
    private String visibility;    // public / unlisted / private
    private boolean isPrivate;

    private Long publishedAt;     // timestamp
    private Long createdAt;       // timestamp
    private Long updatedAt;       // timestamp

    @PrePersist
    public void onCreate() {
        this.createdAt = Instant.now().toEpochMilli();
        this.updatedAt = Instant.now().toEpochMilli();
        if (this.status == null) this.status = "draft";
        if (this.visibility == null) this.visibility = "public";
    }

    @PreUpdate
    public void onUpdate() {
        this.updatedAt = Instant.now().toEpochMilli();
    }
}

File: Mobile/app/src/main/java/com/example/myapplication/data/model/Image.kt
package com.example.myapplication.data.model

data class Image(
    val id: Long? = null,
    val userId: Long? = null,
//    val title: String? = null,
//    val description: String? = null,
    val url: String? = null,
    val status: String? = null,       // draft / processing / published / private / deleted / blocked
    val visibility: String? = null,   // public / unlisted / private
    val isPrivate: Boolean? = false,
    val publishedAt: Long? = null,    // timestamp
    val createdAt: Long? = null,      // timestamp
    val updatedAt: Long? = null       // timestamp
)
